src/main/java/com/bulkbuy/enterprise/BulkBuyController.java

```diff
@@ -15,14 +19,35 @@ public String index() {
         return "start";
     }
 
+    /**
+     * Endpoint handler for /newOrder
+     * @return the page newOrder.html
+     */
     @RequestMapping("/newOrder")
-    public String newOrder(){
+    public String newOrder(Model model){
+        Order order = new Order();
+        model.addAttribute(order);
         return "newOrder";
     }
 
+    /**
+     * Endpoint handler for /orderLookup
+     * @return the page orderLookup.html
+     */
     @RequestMapping("/orderLookup")
-    public String orderLookup(){
+    public String orderLookup(Model model){
+        Order order = new Order();
+        model.addAttribute(order);
         return "orderLookup";
     }
 
+    @RequestMapping("/saveOrder")
+    public String saveOrder(Order newOrder){
+        Date today = new Date();
+        newOrder.setOrderDate(today);
+        newOrder.setOrderStatus("Pending");
+        newOrder.setCarrier("TBD");
+        return "newOrder";
+    }
+
```

Comment:
I've added mapping for saveOrder to handle the submit button on the newOrder page.

Reply:
Should this be calling to a save method on the service?

The method signature (save) and behavior (create and return a new object) are misleading.